### PR TITLE
UIPFO-46: Configure organizations modal to display donor organizations and their `name` and `code` fields only

### PR DESCRIPTION
**Pull request by alisher-epam**
## Purpose

[UIPFO-46](https://issues.folio.org/browse/UIPFO-46) - Modify organizations plugin for donor organization companies and show only `name` and `code` fields.
## Approach

- add `visibleFilters` for displaying desired filters
- allow return all selected vendors when `isMultiSelect` == true

https://github.com/folio-org/ui-plugin-find-organization/assets/116072773/8703a748-9b08-4d2a-bd9c-8996bdc8246a

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: OrganizationSearch/OrganizationsSearchConfig.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ const placeholderIdsMap = {
   [indexesMap.language]: 'ui-organizations.search.placeholder.language',
 };
 
-export const searchableIndexes = [
```
Comment (usavkov-epam) [inline OrganizationSearch/OrganizationsSearchConfig.js:20]: It's a breaking change. This is used outside the plugin.

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,7 @@
 ## (5.1.0 IN PROGRESS)
 
 * Add the "Is Donor" filter selectable options for filters pane. Refs UIPFO-44.
+* Create or add list of donors. Refs UIPFO-46.
```
Comment (usavkov-epam) [inline CHANGELOG.md:6]: Please update the description according to the changes. Currently, it is not entirely clear what has changed in the module.

Comment (usavkov-epam) [reply CHANGELOG.md:6]: And the PR's title as well

File: OrganizationSearch/FindOrganization.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import React, { useState, useCallback } from 'react';
+import React, { useState, useCallback, useMemo } from 'react';
```
Comment (usavkov-epam) [inline OrganizationSearch/FindOrganization.js:1]: code smell (unused import)

File: OrganizationSearch/hooks/useDonors/useDonors.js
Comment (usavkov-epam) [inline OrganizationSearch/hooks/useDonors/useDonors.js:1]: Have you tried to use `makeQueryBuilder` to build the query?

E.g.
https://github.com/folio-org/ui-plugin-find-organization/blob/master/OrganizationSearch/hooks/useOrganizations/useOrganizations.js#L19C20-L30

File: OrganizationSearch/FindOrganization.js [hunk 4]
```diff
@@ -7,18 +7,28 @@ import {
   PLUGIN_RESULT_COUNT_INCREMENT,
 } from '@folio/stripes-acq-components';
 
+import {
+  DEFAULT_VISIBLE_COLUMNS,
+  DONOR_COLUMNS,
+} from './constants';
+import {
+  useDonors,
```
Comment (NikitaSedyx) [inline OrganizationSearch/FindOrganization.js:15]: donor is organization

**Comment by NikitaSedyx** [comment]
you should introduce plugin configuration and not donor changes here

File: OrganizationSearch/FindOrganization.js [hunk 5]
```diff
@@ -7,18 +7,26 @@ import {
...
+  searchableIndexes as organizationSearchableIndexes,
+} from './OrganizationsSearchConfig';
 
 const INIT_PAGINATION = { limit: PLUGIN_RESULT_COUNT_INCREMENT, offset: 0 };
 
 const idPrefix = 'ui-plugin-find-organization-';
-const modalLabel = <FormattedMessage id="ui-plugin-find-organization.modal.label" />;
-const resultsPaneTitle = <FormattedMessage id="ui-plugin-find-organization.meta.pluginTitle" />;
+const defaultModalLabel = <FormattedMessage id="ui-plugin-find-organization.modal.label" />;
+const donorsModalLabel = <FormattedMessage id="ui-plugin-find-organization.donors.modal.label" />;
```
Comment (NikitaSedyx) [inline OrganizationSearch/FindOrganization.js:26]: again, please apply configuration for plugin, it shouldn't know about donor or not donor

File: OrganizationSearch/FindOrganization.js [hunk 6]
```diff
@@ -40,6 +48,12 @@ export const FindOrganization = ({ selectVendor, ...rest }) => {
 
   const { fetchOrganizations } = useOrganizations();
 
+  const searchableIndexes = isDonorsEnabled ? donorsSearchableIndexes : organizationSearchableIndexes;
```
Comment (NikitaSedyx) [inline OrganizationSearch/FindOrganization.js:51]: all of these should come from props - with default values

Comment (alisher-epam) [reply OrganizationSearch/FindOrganization.js:51]: Got it.

**Review by usavkov-epam — APPROVED** [review]
LGTM

**Review by OleksandrHladchenko1 — APPROVED** [review]

**Pull request merged**